src/libraries/System.Security.Cryptography.Cose/tests/CoseSign1MessageTests.Sign.cs

```diff
@@ -175,60 +177,47 @@ public void SignWithIsDetached(bool isDetached)
         {
             byte[] content = GetDummyContent(ContentTestCase.Small);
 
-            byte[] messageEncoded = CoseSign1Message.Sign(content, DefaultKey, DefaultHash, isDetached);
-            VerifyContentDetached(messageEncoded, content, isDetached);
-
-            messageEncoded = CoseSign1Message.Sign(content, DefaultKey, DefaultHash, isDetached);
-            VerifyContentDetached(messageEncoded, content, isDetached);
+            ReadOnlySpan<byte> messageEncoded = CoseSign1Message.Sign(content, DefaultKey, DefaultHash, isDetached: isDetached);
```

Comment:
I don't quite understand the tests that are mixing the abstract Sign with a specific Sign...

Reply:
The problem with `Sign(ECDsa, ...)` and `Sign(RSA, ...)` is that they can't take [un]protected headers, so not all tests apply to them...

I think I can re-shape the tests classes so they can look something like this:

```cs 
    public abstract class CoseSign1MessageTests_Sign<T> where T : AsymmetricAlgorithm
    {
        internal abstract byte[] Sign(byte[] content, T key, HashAlgorithmName hashAlgorithm, bool isDetached = false);

        // Write tests that apply to all [Try]Sign overloads
    }

    public class CoseSign1MessageTests_Sign_ECDsa : CoseSign1MessageTests_Sign<ECDsa>
    {
        internal override byte[] Sign(byte[] content, ECDsa key, HashAlgorithmName hashAlgorithm, bool isDetached = false)
            => CoseSign1Message.Sign(content, key, hashAlgorithm, isDetached);
    }

    public class CoseSign1MessageTests_Sign_RSA: CoseSign1MessageTests_Sign<RSA>
    {
        internal override byte[] Sign(byte[] content, RSA key, HashAlgorithmName hashAlgorithm, bool isDetached = false)
            => CoseSign1Message.Sign(content, key, hashAlgorithm, isDetached);
    }

    public abstract class CoseSign1MessageTests_Sign_WithCustomHeaders : CoseSign1MessageTests_Sign<AsymmetricAlgorithm>
    {
        internal abstract byte[] Sign(byte[] content, AsymmetricAlgorithm key, HashAlgorithmName hashAlgorithm, CoseHeaderMap? protectedHeaders = null, CoseHeaderMap? unprotectedHeaders = null, bool isDetached = false);

        // Write tests that apply only to custom header scenarios
    }

    public class CoseSign1MessageTests_Sign_ByteArray : CoseSign1MessageTests_Sign_WithCustomHeaders
    {
        // proper overrides
    }

    public class CoseSign1MessageTests_Sign_TrySign : CoseSign1MessageTests_Sign_WithCustomHeaders
    {
        // proper overrides
    }
```